packages/cadl-python/test/generated/inheritance/models/inheritance/models/_models.py

```diff
@@ -100,15 +105,15 @@ class Fish(_model_base.Model):
 
     :ivar age: Required.
     :vartype age: int
-    :ivar kind: Required. Default value is "salmon".
+    :ivar kind: Required. Default value is None.
     :vartype kind: str
     """
 
     __mapping__ = {}
     age: int = rest_field()
     """Required. """
-    kind: str = rest_discriminator()
-    """Required. Default value is \"salmon\"."""
+    kind: Literal[None] = rest_discriminator(name="kind")
```

Comment:
have to add `name="kind"` here explicitly after declaring `kind` as `rest_discriminator` in each of the child classes.